crates/volta-core/src/tool/yarn/resolve.rs

```diff
@@ -69,8 +71,8 @@ fn resolve_semver(matching: VersionReq, hooks: Option<&ToolHooks<Yarn>>) -> Fall
     }
 }
 
-fn fetch_yarn_index() -> Fallible<(String, PackageIndex)> {
```

Comment:
Super minor, but do we have a similar function in the `npm` or `package` resolve modules? It may make sense to consolidate them into a single function and then add the `registry_fetch_error` context when calling.

Reply:
There's also a `fetch_npm_index` function that overlaps with a lot of the same fetching code, although it has some hook logic first. So I refactored the duplicated code into a new `fetch_public_index` function, which is called by both of the fetch_*_index` functions, in db9e6d1d0e1a2ca910231af66ce65b8f9416f382